python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,11 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_should_validate_input_args(self):
+        from pyspark.sql.functions import udf
+
+        self.assertRaises(TypeError, udf(lambda x: x), None)
```

Comment:
It is pretty well covered by existing `udf` tests. The more the merrier but I am not sure what can be added with duplicating other test cases.

Do you think we should try to some type validation of the number of arguments?

Pros:
- It is easy to implement with `inspect` or `func.__code__` for plain Python objects. 
- It is nice to fail without starting a complex job.

Cons:
- It most likely won't work well for C extensions and such.